### PR TITLE
Fix tutorials

### PR DESCRIPTION
**Pull request by duboism**
I have started to correct some tutorials. Don't hesitate to suggest other improvements.

### CODE REVIEW

File: doc/devel/matlab_interface_devel.rst [hunk 1]
```diff
@@ -65,66 +68,58 @@ Example 2
...
+    from nipype.interfaces.base import traits
+    from nipype.interfaces.matlab import MatlabCommand, MatlabInputSpec
+    
+    
+    class HelloWorldInputSpec(MatlabInputSpec):
+        name = traits.Str(mandatory=True,
+                          desc='Name of person to say hello to')
+    
+    
+    class HelloWorldOutputSpec(MatlabInputSpec):
```
Comment (satra) [inline doc/devel/matlab_interface_devel.rst]: perhaps this should just be `TraitedSpec` like example 1. does this also answer your question on neurostars?

Comment (duboism) [inline doc/devel/matlab_interface_devel.rst]: I will try that ASAP (and it answers my question on neurostars).

Comment (duboism) [inline doc/devel/matlab_interface_devel.rst]: Is it OK to put the python source code in a separate file (for example `matlab_example2.py`) and include it with `literalinclude`? AFAI can see, the ouput is the same. Moreover we could offer the option to download it as it is done in the `smri_ants_build_template` example.

Comment (satra) [inline doc/devel/matlab_interface_devel.rst]: that's sounds good. all of those are done using py files that get converted from python to rst.

Comment (duboism) [inline doc/devel/matlab_interface_devel.rst]: OK. Therefore I will move the 2 examples here to files.

**Comment by duboism** [comment]
I have meet some problems related to sphinx. Could you have a look at [this question](https://neurostars.org/p/2906/)

**Comment by duboism** [comment]
I have met another sphinx related errors. See the message (here)[https://neurostars.org/p/2936/].

**Comment by nicholsn** [comment]
@duboism, let's continue the technical discussion from http://neurostars.org/p/2936/#2948 here rather than on neurostars.

re: using single line comments vs triple quotes, I like the way the triple quotes break up the documentation into manageable chunks, but as you point out, the compromise is a lack of indentation...

I'd opt for triple quotes, but perhaps @satra or others have a best practice for this

**Comment by duboism** [comment]
I agrre we have to found a tradeoff. I'd opt for 3 comment lines inside functions: I often do that to document large portions of code and users will rely on syntax-highlighting and indentation to catch the structure.

**Comment by duboism** [comment]
@satra : I have merged modifications in `fmri_ants_openfmi.py` but I could not try it. It's mainly PEP8 edits.
@nicholsn :  doc generated from `fmri_ants_openfmi.py` still have syntax-highlighting issues around `get_aparc_aseg`. I haven't been able to fix those.

**Comment by duboism** [comment]
@satra: I won't have much time to work on nipype in the next months so maybe it's better to pull this now.

**Comment by nicholsn** [comment]
@duboism, took a look at `fmri_ants_openfmi.py` and couldn't get it to work. Agree to merge these fixes and maybe raise an issue to get back to this later.

**Comment by duboism** [comment]
@nicholsn Do you mean the test don't run correctly or that the doc still has issues? In the first case, I had problem too (see [here](https://neurostars.org/p/2957/)). Let's hope it will be fixed. In the second case, could you spot the errors?

**Comment by nicholsn** [comment]
The tests are fine, but the docs are still not rendering correctly. I was
unable to spot the error - may need a fresh set of eyes

**Comment by chrisfilo** [comment]
We made some fixes to the docs and added tests for documentation building. Could you rebase?

**Comment by nicholsn** [comment]
Yup, will do

**Comment by chrisfilo** [comment]
any luck?

**Comment by duboism** [comment]
Hi. I'm working on rebasing the branch. There are a lot of conflicts (I haven't touched it since March). I hope to finish today.

**Comment by chrisfilo** [comment]
Excellent thanks!

**Comment by duboism** [comment]
The rebase should be finished. I hope I haven't destroyed any other work in correcting the many conflicts. Don't hesitate to warn me in this case (but I can work on nipype only during my spare time).

File: .travis.yml [hunk 2]
```diff
@@ -42,8 +42,9 @@ deploy:
   provider: pypi
   user: satra
   password:
-    secure: ZaN6Y4hfybPMi1rW3Qe03irCe/3GU4J4BJsjG+tClVN5Fjc3FCJaYU0k4wNd9q2kMjuRDv1efA9BN/5QasJS32oqs6wHvxzIWA18ucel4BBnAAkuviC1vqi/8Hk/DGB6e1gnmZ1Nv58zVLfn0MwlOKmsGDtvGQWc+HbaV2fS5NA=
+    secure: OCO0FXb4f+pH4Uw7zWCIRp3qOJ1t7rhky4K8MjNU8tyVCJgd6O/Bv8GJgceS0LktPodlAAjB8SxAhTORPAQZ1D/44PJYy3NQIisvej1zjLpaA9TEGfl6W7MqhDpRyMHW+cnSi/n84SAmdr+Z4vOxScDHdwr13EPmGyOIlHMAGnE=
```
Comment (chrisfilo) [inline .travis.yml]: was this an accidental change?

File: doc/devel/cmd_interface_devel.rst [hunk 3]
```diff
@@ -39,7 +39,11 @@ above example we have used the ``desc`` metadata which holds human readable
 description of the input. The ``mandatory`` flag forces Nipype to throw an
 exception if the input was not set. ``exists`` is a special flag that works only
 for ``File traits`` and checks if the provided file exists. More details can be
+<<<<<<< HEAD
 found at :doc:`interface_specs`.
+=======
+found at :ref:`interface_specs`.
+>>>>>>> cb80e24fc2a68758defcb16c7ab70092aa35b693
```
Comment (chrisfilo) [inline doc/devel/cmd_interface_devel.rst]: left over?

File: doc/devel/matlab_interface_devel.rst [hunk 4]
```diff
@@ -4,127 +4,39 @@
...
+<<<<<<< HEAD
 
+.. literalinclude:: matlab_example2.py
 
+.. admonition:: Example source code
 
+  You can download :download:`the source code of this example <matlab_example2.py>`.
 
+.. include:: ../links_names.txt
+=======
```
Comment (chrisfilo) [inline doc/devel/matlab_interface_devel.rst]: unresolved conflict?

File: doc/users/tutorial_101.rst [hunk 5]
```diff
@@ -25,31 +25,29 @@ setting up a workflow is separate from executing it.
...
 .. testcode::
-
-   realigner = pe.Node(interface=spm.Realign(infile='somefuncrun.nii',
+   
+   realigner = pe.Node(interface=spm.Realign(infile=os.abspath('somefuncrun.nii'),
+<<<<<<< HEAD
                                              register_to_mean = True),
+=======
+                                             register_to_mean = True), 
+>>>>>>> cb80e24fc2a68758defcb16c7ab70092aa35b693
```
Comment (chrisfilo) [inline doc/users/tutorial_101.rst]: more left over?

**Comment by chrisfilo** [comment]
Thanks. I started going through it but I found many unresolved conflict markers in the code. I think something must've gone wrong when you were resolving merge conflicts.

**Comment by duboism** [comment]
Thanks for the review (I haven't found any tool to help during the rebase so I edited the files manually hence the leftovers and various glitches). I will fix those ASAP.

File: .travis.yml (hunk 2 above)
Comment (duboism) [inline .travis.yml]: I don' remember having touched that file. So probably an accident. I will reverted this file.

Comment (satra) [inline .travis.yml]: this is not comparing properly with master - those changes are already in there. @duboism - could you please close this pull request and send a new one?

Comment (duboism) [inline .travis.yml]: @satra Are you talking about the file .travis.yml or about all the modifications?

Comment (duboism) [inline .travis.yml]: Hum, I have just cloned master in a separate directory and the .travis.yml file is identical the the one in this branch.

**Comment by duboism** [comment]
The leftovers you have pointed should be corrected now.

File: doc/users/tutorial_101.rst [hunk 6]
```diff
@@ -99,16 +93,20 @@ This results in a workflow containing two isolated nodes:
...
+<<<<<<< HEAD
 We want to connect the output produced by the node realignment to the input of
-the node smoothing. This is done as follows.
-
-.. testcode::
+the node smoothing. This is done as follows:
+=======
+We want to connect the output produced by realignment to the input of
+smoothing. This is done as follows:
+>>>>>>> cb80e24fc2a68758defcb16c7ab70092aa35b693
```
Comment (chrisfilo) [inline doc/users/tutorial_101.rst]: unresolved conflict

File: examples/fmri_ants_openfmri.py [hunk 7]
```diff
@@ -105,16 +112,23 @@ def create_reg_workflow(name='registration'):
         outputspec.anat2target_transform : FLIRT+FNIRT transform
         outputspec.transformed_files : transformed files in target space
         outputspec.transformed_mean : mean image in target space
+<<<<<<< HEAD
+=======
+
+    Example
+    -------
+        See code below
+>>>>>>> cb80e24fc2a68758defcb16c7ab70092aa35b693
```
Comment (chrisfilo) [inline examples/fmri_ants_openfmri.py]: unresolved conflict

File: examples/fmri_ants_openfmri.py [hunk 8]
```diff
@@ -236,7 +251,11 @@ def create_reg_workflow(name='registration'):
 
     merge = pe.Node(niu.Merge(2), iterfield=['in2'], name='mergexfm')
     register.connect(convert2itk, 'itk_transform', merge, 'in2')
+<<<<<<< HEAD
     register.connect(reg, 'composite_transform', merge, 'in1')
+=======
+    register.connect(reg, ('composite_transform', pickfirst), merge, 'in1')
+>>>>>>> cb80e24fc2a68758defcb16c7ab70092aa35b693
```
Comment (chrisfilo) [inline examples/fmri_ants_openfmri.py]: unresolved conflict

File: examples/fmri_ants_openfmri.py [hunk 9]
```diff
@@ -286,32 +305,46 @@ def create_reg_workflow(name='registration'):
...
+
 def create_fs_reg_workflow(name='registration'):
     """Create a FEAT preprocessing workflow together with freesurfer
 
     Parameters
     ----------
+<<<<<<< HEAD
 
+=======
+>>>>>>> cb80e24fc2a68758defcb16c7ab70092aa35b693
```
Comment (chrisfilo) [inline examples/fmri_ants_openfmri.py]: unresolved conflict

File: examples/fmri_ants_openfmri.py [hunk 10]
```diff
@@ -286,32 +305,46 @@ def create_reg_workflow(name='registration'):
...
         outputspec.anat2target_transform : FLIRT+FNIRT transform
         outputspec.transformed_files : transformed files in target space
         outputspec.transformed_mean : mean image in target space
+<<<<<<< HEAD
+=======
+
+    Example
+    -------
+        See code below
+>>>>>>> cb80e24fc2a68758defcb16c7ab70092aa35b693
```
Comment (chrisfilo) [inline examples/fmri_ants_openfmri.py]: unresolved conflict

File: examples/fmri_ants_openfmri.py [hunk 11]
```diff
@@ -348,8 +381,13 @@ def create_fs_reg_workflow(name='registration'):
     register.connect(fssource, 'T1', convert, 'in_file')
 
     # Coregister the median to the surface
+<<<<<<< HEAD
     bbregister = Node(freesurfer.BBRegister(registered_file=True),
                     name='bbregister')
+=======
+    bbregister = Node(freesurfer.BBRegister(),
+                      name='bbregister')
+>>>>>>> cb80e24fc2a68758defcb16c7ab70092aa35b693
```
Comment (chrisfilo) [inline examples/fmri_ants_openfmri.py]: unresolved conflict

File: examples/fmri_ants_openfmri.py [hunk 12]
```diff
@@ -435,7 +474,11 @@ def create_fs_reg_workflow(name='registration'):
     reg.plugin_args = {'qsub_args': '-pe orte 4',
                        'sbatch_args': '--mem=6G -c 4'}
     register.connect(stripper, 'out_file', reg, 'moving_image')
+<<<<<<< HEAD
     register.connect(inputnode,'target_image', reg,'fixed_image')
+=======
+    register.connect(inputnode, 'target_image', reg, 'fixed_image')
+>>>>>>> cb80e24fc2a68758defcb16c7ab70092aa35b693
```
Comment (chrisfilo) [inline examples/fmri_ants_openfmri.py]: unresolved conflict

File: examples/fmri_ants_openfmri.py [hunk 13]
```diff
@@ -445,7 +488,11 @@ def create_fs_reg_workflow(name='registration'):
 
     merge = Node(Merge(2), iterfield=['in2'], name='mergexfm')
     register.connect(convert2itk, 'itk_transform', merge, 'in2')
+<<<<<<< HEAD
     register.connect(reg, 'composite_transform', merge, 'in1')
+=======
+    register.connect(reg, ('composite_transform', pickfirst), merge, 'in1')
+>>>>>>> cb80e24fc2a68758defcb16c7ab70092aa35b693
```
Comment (chrisfilo) [inline examples/fmri_ants_openfmri.py]: unresolved conflict

File: examples/rsfmri_vol_surface_preprocessing_nipy.py [hunk 14]
```diff
@@ -505,15 +522,17 @@ def create_reg_workflow(name='registration'):
...
     """
 
     merge = Node(Merge(2), iterfield=['in2'], name='mergexfm')
     register.connect(convert2itk, 'itk_transform', merge, 'in2')
+<<<<<<< HEAD
     register.connect(reg, 'composite_transform', merge, 'in1')
-
+=======
+    register.connect(reg, ('composite_transform', pickfirst), merge, 'in1')
+>>>>>>> cb80e24fc2a68758defcb16c7ab70092aa35b693
```
Comment (chrisfilo) [inline examples/rsfmri_vol_surface_preprocessing_nipy.py]: unresolved conflict

File: examples/rsfmri_vol_surface_preprocessing_nipy.py [hunk 15]
```diff
@@ -606,8 +623,12 @@ def create_workflow(files,
                        name='median')
     wf.connect(tsnr, 'detrended_file', calc_median, 'in_files')
 
-    """Segment and Register
     """
+<<<<<<< HEAD
+=======
+    Segment and Register
+    """
+>>>>>>> cb80e24fc2a68758defcb16c7ab70092aa35b693
```
Comment (chrisfilo) [inline examples/rsfmri_vol_surface_preprocessing_nipy.py]: unresolved conflict

File: examples/rsfmri_vol_surface_preprocessing_nipy.py [hunk 16]
```diff
@@ -624,6 +646,8 @@ def create_workflow(files,
     wf.connect(tsnr, 'tsnr_file', get_roi_tsnr, 'in_file')
     wf.connect(registration, 'outputspec.aparc', get_roi_tsnr, 'segmentation_file')
 
+=======
+>>>>>>> cb80e24fc2a68758defcb16c7ab70092aa35b693
```
Comment (chrisfilo) [inline examples/rsfmri_vol_surface_preprocessing_nipy.py]: unresolved conflict

**Comment by djarecka** [comment]
@duboism - it's a great pr, would you consider finishing it?

**Comment by djarecka** [comment]
@duboism - please let us know if you're planning to work on this PR, otherwise I'll try to review and finish/close.

**Comment by duboism** [comment]
Hello,
I have moved to a completely different job. I may find some time to finish this but it's not sure so feel free to work on the PR.

**Comment by djarecka** [comment]
@duboism - thank you for answering. We will be merging with the Nipype Tutorial, so I will review which changes will be useful in the new verion. Thanks!

**Comment by satra** [comment]
@djarecka - is this something that can be resolved before next release.

**Comment by djarecka** [comment]
I haven't reviewed this yet. Will try to it next week

**Pull request closed without merging**